### PR TITLE
Feature/cvsb 7391

### PR DESCRIPTION
**Pull request by oalexandru**

### CODE REVIEW

File: tests/unit/insertTestResult.unitTest.ts [hunk 1]
```diff
@@ -701,4 +702,461 @@ describe("insertTestResult", () => {
...
+                    expect(error).to.be.instanceOf(HTTPError);
+                    expect(error.statusCode).to.be.eql(400);
+                });
+        });
+    });
+
+    context('when inserting a cancelled HGV that has null values on the fields that are allowing them to be null', () => {
+        it('should not throw error', () => {
+            let testResult = testResultsPostMock[4]
+            testResult.testStatus = 'cancelled'
```
[the remark this answers is not in the document]
Comment (KHodderDVSA) [reply tests/unit/insertTestResult.unitTest.ts]: image

File: tests/unit/insertTestResult.unitTest.ts [hunk 2]
```diff
@@ -701,4 +702,461 @@ describe("insertTestResult", () => {
...
+                        });
+                    }
+                };
+            });
+
+            testResultsService = new TestResultsService(new MockTestResultsDAO());
+
+            return testResultsService.insertTestResult(testResult)
+                .then((insertedTestResult: any) => {
+                    expect(insertedTestResult).to.not.be.eql(undefined)
```
Comment (KHodderDVSA) [inline tests/unit/insertTestResult.unitTest.ts]: bad test - can match with error text/stack trace. If you're expecting something specific, match against that.

File: tests/unit/insertTestResult.unitTest.ts [hunk 3]
```diff
@@ -701,4 +702,461 @@ describe("insertTestResult", () => {
...
+                })
+                .catch(() => {
+                    expect.fail()
+                })
+        })
+    })
+
+    context('when inserting an HGV with fields corresponding to a PSV', () => {
+        it('should throw 400', () => {
+            let testResult = testResultsPostMock[2]
```
Comment (KHodderDVSA) [inline tests/unit/insertTestResult.unitTest.ts]: as above - no mutations of common object

File: tests/unit/insertTestResult.unitTest.ts [hunk 4]
```diff
@@ -701,4 +702,461 @@ describe("insertTestResult", () => {
...
+                        });
+                    }
+                };
+            });
+
+            testResultsService = new TestResultsService(new MockTestResultsDAO());
+
+            return testResultsService.insertTestResult(testResult)
+                .then((data: any) => {
+                    expect(data).to.not.be.eql(undefined)
```
Comment (KHodderDVSA) [inline tests/unit/insertTestResult.unitTest.ts]: as above

File: tests/unit/insertTestResult.unitTest.ts [hunk 5]
```diff
@@ -161,6 +162,7 @@ describe("insertTestResult", () => {
                 })
                 .catch((error: { statusCode: any; body: any; }) => {
                     expect(error).to.be.instanceOf(HTTPError);
+                    console.log(error.body)
```
[the remark this answers is not in the document]
Comment (arupdvsa) [reply tests/unit/insertTestResult.unitTest.ts]: No we do not need to log errors in tests (not expectations) ... fine to do debug of tests ... but once debug done ... should be removed

File: tests/unit/insertTestResult.unitTest.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (arupdvsa) [reply tests/unit/insertTestResult.unitTest.ts]: object spread will do a shallow copy ... and could be issue with deep nested jsons (few issues observed earlier) ... so generally use lodash _cloneDeep instead ... but this suggestion can be deferred until 7968

[the remark this answers is not in the document]
Comment (oalexandru) [reply tests/unit/insertTestResult.unitTest.ts]: JSON.parse(JSON.stringify()) could be enough for this without importing the massive lodash library

File: tests/unit/insertTestResult.unitTest.ts [hunk 6]
```diff
@@ -701,4 +702,461 @@ describe("insertTestResult", () => {
...
+                        });
+                    }
+                };
+            });
+
+            testResultsService = new TestResultsService(new MockTestResultsDAO());
+
+            return testResultsService.insertTestResult(testResult)
+                .then((insertedTestResult: any) => {
+                    expect(insertedTestResult).to.not.be.eql(undefined)
```
[the remark this answers is not in the document]
Comment (oalexandru) [reply tests/unit/insertTestResult.unitTest.ts]: the most important result of the insert would be that it doesn't throw validation errors when inserting and not to test that I get back the same object that I inserted. also we have many generated fields in the backend which I need to include in the expected result, work which did not seem necessary for what I'm testing now. But I can do it if you think it's better

**Comment by arupdvsa** [comment]
we know these PR needs to be resolved and implemented separately

**Pull request merged**